packages/api/src/external/commonwell/command/cw-patient-data/create-cw-data.ts

```diff
@@ -0,0 +1,36 @@
+import { NetworkLink } from "@metriport/commonwell-sdk";
+import { executeOnDBTx } from "../../../../models/transaction-wrapper";
+import { CwPatientData, CwPatientDataCreate } from "../../cw-patient-data";
+import { CwPatientDataModel } from "../../models/cw-patient-data";
+import { getCwPatientData } from "./get-cw-data";
+import { updateCwPatientDataWithinDBTx } from "./update-cw-data";
+
+export async function createOrUpdateCwPatientData({
+  id,
+  cxId,
+  cwLinks,
+}: {
+  id: string;
+  cxId: string;
+  cwLinks: NetworkLink[];
+}): Promise<CwPatientData | undefined> {
+  const cwPatientData: CwPatientDataCreate = {
+    id,
+    cxId,
+    data: { links: cwLinks },
+  };
+
+  const updateResult = await executeOnDBTx(CwPatientDataModel.prototype, async transaction => {
+    const existingPatient = await getCwPatientData({
+      id,
+      cxId,
+      transaction,
+    });
```

Comment:
Here and on `updateCwPatientData`, I think we should pass a `lock: true` param so the record is locked for updates. That's the only reason I an think of why it justifies having a DB tx on these commands - and it makes sense considering this can be updated multiple times asynchronously.

Prob a good idea to replicate this to the respective CQ code as well.

Reply:
Done but please verify to make sure I understood you correctly.

Reply:
We should also add it to `updateCwPatientData` - it calls `getCwPatientDataOrFail`. And probably to `deleteCwPatientData` as well?

Reply:
Done.